File: scripts/algorithms/M/Minimum Number of Operations to Convert Time/Minimum Number of Operations to Convert Time.cpp
class Solution {
public:
    int HHMMToMinutes(string s){
        return stoi(s.substr(0,2))*60 + stoi(s.substr(3,2));
    }
    int convertTime(string current, string correct) {
        int diff = - HHMMToMinutes(current) +  HHMMToMinutes(correct);
        vector<int> order = {60,15,5,1};
        int i = 0;
        int ans = 0;
        while(i < 4){
            ans+=(diff/order[i]);
            diff%=order[i];
            i++;
        }
        
        return ans;
    }
};

File: scripts/algorithms/M/Minimum Number of Operations to Convert Time/Minimum Number of Operations to Convert Time.java
// Runtime: 2 ms (Top 77.73%) | Memory: 42.8 MB (Top 23.55%)
class Solution {
    public int HHMMToMinutes(String s){
        return Integer.parseInt(s.substring(0,2))*60 + Integer.parseInt(s.substring(3,5)) ;
    }
    public int convertTime(String current, String correct) {
        int diff = HHMMToMinutes(correct) - HHMMToMinutes(current);
        int[] order = {60,15,5,1};
        int i = 0;
        int ops = 0;
        while(i < 4){
            ops+=(diff/order[i]);
            diff%=order[i];
            i++;
        }
        return ops;
    }
}